reconcile/openshift_resources_base.py

```diff
@@ -300,6 +300,19 @@ def lookup_graphql_query_results(query: str, **kwargs) -> list[Any]:
     return results
 
 
+def hash_list(input: Iterable) -> str:
+    """
+    Deterministic hash of a list for jinja2 templates.
+    The order of the list doesnt matter as it is sorted
+    before hashing. Note, that the list elements
+    must be flat (no dicts).
```

Comment:
What happens if you pass a dict? Shouldn't it raise a well-known exception?

Reply:
Very good point. Added a check that we only work with primitives. Added some tests for the function too.